docs/source/graphs.rst

```diff
@@ -97,3 +97,39 @@ special and others can write different schedulers better suited to other
 applications or architectures easily.  Systems that emit dask graphs (like
 Dask Array, Dask Bag, and so on) may leverage the appropriate scheduler for
 the application and hardware.
+
+
+Task Expectations
+-----------------
+
+When a task is submitted to Dask for execution, there are a number of assumptions
+that are made about that task. In general, tasks with side-effects that alter the
+state of a future in-place are not recommended. Dask Futures are mutable, and can
+be updated in-place. For example, consider a workflow involving a
+``pandas.DataFrame``:
+
+.. code-block:: python
+
+   from distributed import Client
```

Comment:
We prefer to use the namespaced term `dask.distributed`.  In hindsight, naming a library with a generic term like `distributed` was a terrible choice :)

```suggestion
   from dask.distributed import Client
```